From 19b070051c1c2b5fb6f2da8fb6422b27418c3dfa Mon Sep 17 00:00:00 2001
From: Kefu Chai <tchaikov@gmail.com>
Date: Mon, 15 Jul 2024 09:27:16 +0800
Subject: [PATCH] blocks,runtime: io_signature: include spdlog/*/ranges.h for
 using fmt::join()

fmt::join() was moved into fmt/ranges.h since fmt 11, so let's
include the corresponding header in spdlog for using it.

Signed-off-by: Kefu Chai <tchaikov@gmail.com>
---
 gnuradio-runtime/lib/io_signature.cc | 5 +++++
 gr-blocks/lib/message_debug_impl.cc  | 5 +++++
 2 files changed, 10 insertions(+)

diff --git a/gnuradio-runtime/lib/io_signature.cc b/gnuradio-runtime/lib/io_signature.cc
index 812f444f8bb..76bb2aba670 100644
--- a/gnuradio-runtime/lib/io_signature.cc
+++ b/gnuradio-runtime/lib/io_signature.cc
@@ -16,6 +16,11 @@
 #include <spdlog/tweakme.h>
 
 #include <spdlog/fmt/fmt.h>
+#if __has_include(<spdlog/fmt/ranges.h>)
+#include <spdlog/fmt/ranges.h>
+#elif __has_include(<spdlog/fmt/bundled/ranges.h>)
+#include <spdlog/fmt/bundled/ranges.h>
+#endif
 #include <string_view>
 #include <algorithm>
 #include <memory>
diff --git a/gr-blocks/lib/message_debug_impl.cc b/gr-blocks/lib/message_debug_impl.cc
index 41d312e43c3..373287b02ab 100644
--- a/gr-blocks/lib/message_debug_impl.cc
+++ b/gr-blocks/lib/message_debug_impl.cc
@@ -19,6 +19,11 @@
 #include <pmt/pmt.h>
 #include <spdlog/common.h>
 #include <spdlog/fmt/fmt.h>
+#if __has_include(<spdlog/fmt/ranges.h>)
+#include <spdlog/fmt/ranges.h>
+#elif __has_include(<spdlog/fmt/bundled/ranges.h>)
+#include <spdlog/fmt/bundled/ranges.h>
+#endif
 #include <functional>
 #include <utility>
 #include <vector>
